### PR TITLE
python3.pkgs.django-filter: 23.2 -> 23.3

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yu-re-ka** [comment]
Built nixosTests.peering-manager

File: pkgs/development/python-modules/django-filter/default.nix [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
 , fetchPypi
+, flit-core
 , django
 , djangorestframework, python, mock
 }:
 
 buildPythonPackage rec {
   pname = "django-filter";
-  version = "23.2";
+  version = "23.3";
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-filter/default.nix:11]: format right below version, please.

File: pkgs/development/python-modules/django-filter/default.nix [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-  # Tests fail (needs the 'crispy_forms' module not packaged on nixos)
-  doCheck = false;
-
   nativeCheckInputs = [
     djangorestframework
-    django
-    mock
   ];
 
   checkPhase = ''
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-filter/default.nix:32]: the homepage is https://github.com/carltongibson/django-filter
the changelog is here `https://github.com/carltongibson/django-filter/blob/v${version}/CHANGES.rst`

File: pkgs/development/python-modules/django-filter/default.nix [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
   ];
 
-  # Tests fail (needs the 'crispy_forms' module not packaged on nixos)
-  doCheck = false;
-
   nativeCheckInputs = [
     djangorestframework
-    django
-    mock
   ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-filter/default.nix:30]: Use pytestCheckHook, pytest-django for better control when running the test suite.

Requires exporting the following env var

> DJANGO_SETTINGS_MODULE", "tests.settings"

Drop the explicit checkPhase after that.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/django-filter/default.nix (hunk 1 above)
Comment (yu-re-ka) [reply pkgs/development/python-modules/django-filter/default.nix:14]: Done

File: pkgs/development/python-modules/django-filter/default.nix (hunk 2 above)
Comment (yu-re-ka) [reply pkgs/development/python-modules/django-filter/default.nix]: Done

File: pkgs/development/python-modules/django-filter/default.nix (hunk 3 above)
Comment (yu-re-ka) [reply pkgs/development/python-modules/django-filter/default.nix]: Done

File: pkgs/development/python-modules/django-filter/default.nix [hunk 4]
```diff
@@ -1,43 +1,44 @@
...
+    pytestCheckHook
+    pytest-django
   ];
 
-  checkPhase = ''
-    runHook preCheck
-    ${python.interpreter} runtests.py tests
-    runHook postCheck
-  '';
+  DJANGO_SETTINGS_MODULE = "tests.settings";
```
Comment (mweinelt) [inline pkgs/development/python-modules/django-filter/default.nix:36]:
```suggestion
  env.DJANGO_SETTINGS_MODULE = "tests.settings";
```

**Pull request merged**